manifests/http_port.pp

```diff
@@ -1,24 +1,56 @@
 define squid::http_port (
-  Variant[Pattern[/\d+/], Integer]
-          $port    = $title,
-  Boolean $ssl     = false,
-  String  $options = '',
-  String  $order   = '05',
+  Optional[Stdlib::Port] $port    = undef,
+  Optional[Stdlib::Host] $host    = undef,
+  Boolean                $ssl     = false,
+  String                 $options = '',
+  String                 $order   = '05',
 ) {
+  $_title = String($title)
+
+  # Try to extract host/port from title if neither were specified as
+  # parameters. Allowed formats: host:port and port.
+  if $host == undef and $port == undef and $_title =~ /^(?:(.+):)?(\d+)$/ {
+    $_host = $1
+    if $_host != undef and $_host !~ Stdlib::Host {
```

Comment:
Nit: You can actually simplify this to `if $_host !~ Optional[Stdlib::Host] {`. You could use [assert_type](https://puppet.com/docs/puppet/5.3/function.html#asserttype) as well but I think it's actually less readable here.

Reply:
Oops, good point about `Optional`. Fixed.

Reply:
Oops, good point about Optional. Fixed.

PS: the tests failed because `The program 'bundle' is currently not installed`.